app/vmselect/netstorage/tmp_blocks_file.go

```diff
@@ -35,14 +40,15 @@ func maxInmemoryTmpBlocksFile() int {
 		return 64 * 1024
 	}
 	if maxLen > 4*1024*1024 {
-		return 4 * 1024 * 1024
+		return *tmpBufSize
 	}
 	return maxLen
 }
 
 var _ = metrics.NewGauge(`vm_tmp_blocks_max_inmemory_file_size_bytes`, func() float64 {
 	return float64(maxInmemoryTmpBlocksFile())
 })
+var tmpBufSizeSummary = metrics.NewSummary(`vm_select_tmp_buffer_size_used`)
```

Comment:
I suggest to rename metric 

`vm_select_tmp_buffer_size_used` to `vm_tmp_blocks_inmemory_file_size_bytes`